subsys/zigbee/lib/zigbee_fota/src/dfu_multi_target.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log.h>
+#include <dfu/dfu_multi_image.h>
+#include <dfu/mcuboot.h>
+#include <dfu/dfu_target.h>
+#include <dfu/dfu_target_mcuboot.h>
+#include <pm_config.h>
+#include "dfu_multi_target.h"
+
+union dfu_multi_target_ver {
+	uint32_t uint32_ver;
+	struct {
+		uint16_t revision;
+		uint8_t minor;
+		uint8_t major;
+	};
+};
+
+static int dfu_multi_target_open(int image_id, size_t image_size);
+static int dfu_multi_target_write(const uint8_t *chunk, size_t chunk_size);
+static int dfu_multi_target_done(bool success);
+
+LOG_MODULE_DECLARE(zigbee_fota);
+
+/** Buffer for parsing the multi-image header as well as DFU target buffer. */
+static uint8_t image_buf[DFU_MULTI_TARGET_BUFFER_SIZE] __aligned(4);
+
+/** DFU target writer for image 0 (single-app image or APP core). */
+static struct dfu_image_writer dfu_multi_target_writer_0 = {
+	.image_id = 0,
+	.open = dfu_multi_target_open,
+	.write = dfu_multi_target_write,
+	.close = dfu_multi_target_done
+};
+
+/** DFU target writer for image 1 (NET core). */
+static struct dfu_image_writer dfu_multi_target_writer_1 = {
```

Comment:
This should be conditionally enabled, based on the `CONFIG_UPDATEABLE_IMAGE_NUMBER` value.